src/controllers/EvidenceUploadController.ts

```diff
@@ -24,36 +32,100 @@ const navigation = {
     }
 };
 
+const continueButtonValidator: Validator = {
+
+    validate(request: Request): ValidationResult {
+
+        const appeal: Appeal = request.session
+            .chain(_ => _.getExtraData())
+            .map<ApplicationData>(data => data[APPLICATION_DATA_KEY])
+            .map(data => data.appeal)
+            .unsafeCoerce();
+
+        const attachments: Attachment[] | undefined = appeal.reasons.other.attachments;
+
+        if (!attachments || attachments.length === 0) {
+            return new ValidationResult([new ValidationError('file',
+                'You must add a document or click “Continue without adding documents”')]);
+        }
+        return new ValidationResult([]);
+    }
+};
+
 @controller(EVIDENCE_UPLOAD_PAGE_URI, SessionMiddleware, AuthMiddleware, FileTransferFeatureMiddleware)
 export class EvidenceUploadController extends BaseController<OtherReason> {
     constructor(@inject(FileTransferService) private readonly fileTransferService: FileTransferService) {
-        super(template, navigation);
+        super(template, navigation, continueButtonValidator);
     }
 
     protected prepareViewModelFromAppeal(appeal: Appeal): Record<string, any> & OtherReason {
         return appeal.reasons?.other;
     }
 
+    private async renderUploadError(appeal: Appeal, text: string): Promise<void> {
+        const validationResult: ValidationResult = new ValidationResult([
+            new ValidationError('file', text)
+        ]);
+
+        return await this.renderWithStatus(UNPROCESSABLE_ENTITY)(
+            this.template, {
+                ...this.prepareViewModelFromAppeal(appeal),
+                ...this.httpContext.request.body,
+                validationResult
+            }
+        );
+    }
+
     protected getExtraActionHandlers(): Record<string, FormActionHandler | FormActionHandlerConstructor> {
         const that = this;
+
+        const fileTooLargeError: string = 'File size must be smaller than 4MB';
+        const fileNotSupportedError: string = 'The selected file must be a TXT, DOC, PDF, JPEG or PNG';
+        const tooManyFilesError: string = 'You can only select up to 10 files at the same time';
```

Comment:
I guess error message might reuse `MAX_NUMBER_OF_FILES ` env variable as this is simple interpolation here.

Reply:
Or just add it to tech debt as the other thing.